pkg/controller/install/certresources.go

```diff
@@ -246,7 +246,7 @@ func (i *StrategyDeploymentInstaller) installCertRequirementsForDeployment(deplo
 
 		// Delete the Service to replace
 		deleteErr := i.strategyClient.GetOpClient().DeleteService(service.GetNamespace(), service.GetName(), &metav1.DeleteOptions{})
-		if err != nil && !k8serrors.IsNotFound(deleteErr) {
```

Comment:
This looks like something that should have been picked up by unit testing. Would it be a lot of effort to add a test?

Reply:
Since e2e is green, let's merge and put the unit test in a follow up PR. Would that work for you?

Reply:
Yes, although I'm not sure how can we check this particular case in a unit test, in which the service could not be deleted and therefore the `installCertRequirementsForDeployment` method returns an error.
I assume we'll need to use:
```
mockOpClient.EXPECT().DeleteService(namespace, "test-service", &metav1.DeleteOptions{}).Return(fmt.Errorf("could not delete existing service test-service"))
```
But not quite sure how to proceed.